test/jdk/java/lang/ProcessBuilder/ReaderWriterTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import static org.testng.Assert.*;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import jdk.test.lib.process.ProcessTools;
+
+import jdk.test.lib.hexdump.HexPrinter;
+import jdk.test.lib.hexdump.HexPrinter.Formatters;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Locale;
+
+/*
+ * @test
+ * @library /test/lib
+ * @build jdk.test.lib.process.ProcessTools jdk.test.lib.hexdump.HexPrinter
+ * @run testng ReaderWriterTest
+ */
+
+@Test
+public class ReaderWriterTest {
+
+    static final String ASCII = "ASCII: \u0000_A-Z_a-Z_\u007C_\u007D_\u007E_\u007F_;";
+    static final String ISO_8859_1 = " Symbols: \u00AB_\u00BB_\u00fc_\u00fd_\u00fe_\u00ff;";
+    static final String FRACTIONS = " Fractions: \u00bc_\u00bd_\u00be_\u00bf;";
+
+    public static final String TESTCHARS = "OneWay: " + ASCII + ISO_8859_1 + FRACTIONS;
+    public static final String ROUND_TRIP_TESTCHARS = "RoundTrip: " + ASCII + ISO_8859_1 + FRACTIONS;
+
+    @DataProvider(name="CharsetCases")
+    static Object[][] charsetCases() {
+        return new Object[][] {
+                {"UTF-8"},
+                {"ISO8859-1"},
```

Comment:
`ISO8859-1` may not be available on all underlying OSes.

Reply:
Is there a safe subset?
I haven't seen a failure yet, if/when it occurs, we make an exception or narrow the test to known systems.